pymc/distributions/discrete.py

```diff
@@ -1423,9 +1423,14 @@ def _zero_inflated_mixture(*, name, nonzero_p, nonzero_dist, **kwargs):
         nonzero_dist,
     ]
     if name is not None:
-        return Mixture(name, weights, comp_dists, **kwargs)
+        rv_out = Mixture(name, weights, comp_dists, **kwargs)
     else:
-        return Mixture.dist(weights, comp_dists, **kwargs)
+        rv_out = Mixture.dist(weights, comp_dists, **kwargs)
+
+    # overriding Mixture _print_name
+    set_print_name(cls, rv_out)
```

Comment:
The alternative would be to be more clever about mixture print name and on the spot check:
1. If it has 2 components and one is a zero constant, call it "ZeroInflatedX", where "X" is the name of the nonzero component, like "ZeroInflatedBinomial"
2. If all the components follow the same distribution call it "XMixture" like "NormalMixture" or "GammaMixture"
3. If there are two different components, call it "X-YMixture" like "Gamma-ExponentialMixture"
4. Otherwise call it simply Mixture

This is where the dispatching idea becomes a bit more powerful. You can have an arbitrarily complex function that specializes on an `Op` and can look at its inputs at evaluation time to figure out a nice name.

The same type of logic could be used for Censored (dispatched on at.clip) to call it "CensoredX" like "CensoredNormal", or the future RandomWalk to call them "XRandomWalk" (perhaps with a special case for Normal, where we call it "Gaussian")

The dispatching itself only means we don't need to define the name at creation time (and others can overwrite it more easily). The more fundamental difference is that it uses a per-op function to decide what name to give to the distribution. 

The basecase would be the `RandomVariable` `Op` which does what was already done eagerly before this PR.

This is just an idea. Feel free to investigate something like this or leave it as a separate enhancement/feature request issue!